feature/gribi/otg_tests/hierarchical_weight_resolution_test/hierarchical_weight_resolution_test.go

```diff
@@ -0,0 +1,739 @@
+	eth.Connection().SetPortName(p.ID())
+	ipObj := eth.Ipv4Addresses().Add().SetName(dev.Name() + ".IPv4")
+	ipObj.SetAddress(ip).SetGateway(gateway).SetPrefix(uint32(a.IPv4Len))
+	t.Logf("Adding ATE Ipv4 address: %s with gateway: %s", cidr(ip, 30), gateway)
+
+	for i := uint32(1); i <= a.numSubIntf; i++ {
+		name := fmt.Sprintf(`dst%d`, i)
+		ip = a.ip(uint8(i))
+		gateway = a.gateway(uint8(i))
+		mac, _ := incrementMAC(a.MAC, int(i)+1)
+
+		dev := top.Devices().Add().SetName(name + ".Dev")
+		eth := dev.Ethernets().Add().SetName(name + ".Eth").SetMac(mac)
+		eth.Connection().SetPortName(p.ID())
+		eth.Vlans().Add().SetName(name).SetId(uint32(i))
+		eth.Ipv4Addresses().Add().SetName(name + ".IPv4").SetAddress(ip).SetGateway(gateway).SetPrefix(uint32(a.IPv4Len))
+
+		t.Logf("Adding ATE Ipv4 address: %s with gateway: %s and VlanID: %d", cidr(ip, 30), gateway, i)
+	}
+}
+
+// incrementMAC increments the MAC by i. Returns error if the mac cannot be parsed or overflows the mac address space
+func incrementMAC(mac string, i int) (string, error) {
+	macAddr, err := net.ParseMAC(mac)
+	if err != nil {
+		return "", err
+	}
+	convMac := binary.BigEndian.Uint64(append([]byte{0, 0}, macAddr...))
+	convMac = convMac + uint64(i)
+	buf := new(bytes.Buffer)
+	err = binary.Write(buf, binary.BigEndian, convMac)
+	if err != nil {
+		return "", err
+	}
+	newMac := net.HardwareAddr(buf.Bytes()[2:8])
+	return newMac.String(), nil
+}
+
+// testTraffic creates a traffic flow with ATE source & destination endpoints
+// and configures a VlanID filter for output frames. The IPv4 header for the
+// flow contains the ATE:Port1 address as source and the configured gRIBI-
+// IndirectEntry as the destination. The function also takes as input a map of
+// <VlanID::TrafficDistribution> that is wanted and compares it to the actual
+// traffic test result.
+func testTraffic(t *testing.T, ate *ondatra.ATEDevice, top gosnappi.Config) map[string]float64 {
+
+	dut := ondatra.DUT(t, "dut")
+	dstMac := gnmi.Get(t, dut, gnmi.OC().Interface(dut.Port(t, "port1").Name()).Ethernet().MacAddress().State())
+	top.Flows().Clear().Items()
+	flowipv4 := top.Flows().Add().SetName("flow")
+	flowipv4.Metrics().SetEnable(true)
+	flowipv4.TxRx().Port().SetTxName(atePort1.Name).SetRxNames([]string{atePort2.Name})
+	flowipv4.Size().SetFixed(100)
+	e1 := flowipv4.Packet().Add().Ethernet()
+	e1.Src().SetValue(atePort1.MAC)
+	e1.Dst().SetValue(dstMac)
+	v4 := flowipv4.Packet().Add().Ipv4()
+	v4.Src().SetValue(atePort1.IPv4)
+	v4.Dst().SetValue(ipv4FlowIP)
+	v4Inner := flowipv4.Packet().Add().Ipv4()
+	v4Inner.Src().Increment().SetStart(innerSrcIPv4Start).SetCount(ipv4FlowCount)
+	v4Inner.Dst().Increment().SetStart(innerDstIPv4Start).SetCount(ipv4FlowCount)
+	flowipv4.EgressPacket().Add().Ethernet()
+	vlan := flowipv4.EgressPacket().Add().Vlan()
+	vlanTag := vlan.Id().MetricTags().Add()
+	vlanTag.SetName("EgressVlanIdTrackingFlow")
+	ate.OTG().PushConfig(t, top)
+	ate.OTG().StartProtocols(t)
```

Comment:
We'll need an `otgutils.WaitForARP` after this.

Reply:
the otgutils.WaitForARP will fail on the receiving side, but the ARP works on the Tx port, so I don't think we need it here

Reply:
We do see flakiness without the `WaitForARP` here.

Reply:
otgutils.WaitForARP readded after removing the extra interface on port2.